### ISSUE TITLE
Wav2Vec 2.0 pretraining stuck

### ISSUE THREAD

**Issue by jubick1337** [pytorch/fairseq#3661]
#### What is your question? 
Why loss and accuracy plots look like this?
<img width="375" alt="image" src="https://user-images.githubusercontent.com/16004982/123698244-5e50e600-d877-11eb-946d-0642ba6c2840.png">
<img width="381" alt="image" src="https://user-images.githubusercontent.com/16004982/123698288-6f99f280-d877-11eb-98e9-8fe58ea11317.png">

#### Code
Running hydra train as it's shown in readme.md for w2v2 base. 
Changes to config:
  restore_file: /root/wav2vec_small.pt                                                                                                                   
  reset_dataloader: true                                                                                                                                 
  reset_lr_scheduler: true                                                                                                                               
  reset_meters: true                                                                                                                                     
  reset_optimizer: true  
  num_workers: 24
  max_tokens: 1500000     
  ddp_backend: no_c10d   
                                                                                                                              
I've parsed hydra_train.log file for my pretraining then plotted it.
#### What have you tried?
.
#### What's your environment?

 - fairseq Version (e.g., 1.0 or master): master
 - PyTorch Version (e.g., 1.0): 1.9.0a0+c3d40fd
 - OS (e.g., Linux): Ubuntu 20.04 LTS
 - How you installed fairseq (`pip`, source): source
 - Build command you used (if compiling from source): pip install --editable ./
 - Python version: 3.8.10
 - CUDA/cuDNN version: 11.3
 - GPU models and configuration: RTX 3090
 - Any other relevant information: It seems like this strange shape occurs once loss_1 and loss_2 become 0

**Comment by jubick1337** [comment]
Went on and checked stable version 0.10.2
Process seems to be the same
Loss freezes at 6.658 in training phase
Although validation is quite nice and validation loss goes to low values as well as accuracy (about 0.02 and 0.99)
Can someone explain this strange thing?
Is it ok to use checkpoint like this? 
P.S. valid set to be 0.01 of original dataset

**Comment by jubick1337** [comment]
One more test to prove there's somethings wrong
valid.tsv == train.tsv
Loss at train goes to 6.658 and right after that valid loss becomes almost 0 and valid accuracy goes to 1
Have anyone else encountered something like that?

**Comment by shiva1393** [comment]
@jubick1337 I'm also getting same problem.

**Comment by jubick1337** [comment]
@shiva1393 do you have any idea how to solve this problem?

**Comment by shiva1393** [comment]
Hi,
   decrease learning rate it will help, this suggestion i got from previous issues.

**Comment by jubick1337** [comment]
@shiva1393 to which values?

**Comment by jubick1337** [comment]
lowering lr to 0.0001 helps

**Comment by bmilde** [comment]
I have the same problem here trying to train a base model from scratch. Also loss shoots up to exactly 6.658 for me, so I think that is some kind of degenerate solution. Accuracy is close to 0 (on train). Will try to lower the learning rate, as suggested by @jubick1337 @shiva1393. I am training on a single GPU, maybe that is making the difference?

**Comment by Etoye** [comment]
@jubick1337 Hi, I know it's an old issue, but how did you solve it? Reduced learning rate, replicated GPUs, or both at the same time?

**Comment by bmilde** [comment]
Hi @Etoye,

I managed to solve it. I had to reduce the learning rate *and* also introduce gradient accumulation (simulating multiple GPUs). Specifically, in the optimization part of the base model config I have:

optimization:
  max_update: 400000
  lr: [0.0002]
  update_freq: [8]

I was able to train a base model successfully with this on just one GPU with fp16. I think the default learning rate was 0.0005.

**Comment by Etoye** [comment]
@bmilde Thank you for the details, it's very helpful! I am wonder if I have 8 GPUs, will it be enough to just replicate 64 GPUs by using update_freq: [8] or reducing learning rate is critical?

**Comment by bmilde** [comment]
Unfortunately its trial and error. If you have 8 GPUs, I'd try update_freq: [8] first. Maybe you can also just lower the learning rate without setting update_freq (this would give you exactly my config).

For me, with just one GPU, setting update_freq: [64] didn't work for whatever reason, but basically simulating 8 GPUs with lower learning rate did converge.
